python/src/aiconfig/eval/README.md

```diff
@@ -0,0 +1,110 @@
+# Tl;Dr
+
+Iterate on AIConfig faster. Measure AIConfig quality improvements and catch regressions. This module is closely related to [Promptfoo](https://www.promptfoo.dev/) and in fact it provides the option to integrate directly with Promptfoo. Please refer to their docs for motivating examples and usage details as needed.
+
+- You specify an AIConfig, test data, and evaluation metrics / assertions.
+- Each test case is a pair of (data, evaluation_metric).
+- Metrics can be anything numerical (float/int) or any boolean condition (pass/fail).
+- AIConfig eval library runs them through your AIConfig and presents you a table to compare the results. Example:
+
+Inputs:
+
+```
+test_inputs = [
+    "Food",
+    "Iconic midtown skyscrapers",
+]
+expected_substrings = [
+    "Magnolia Bakery",
+    "Empire State Building",
+]
+
+# For each input, ask the library to check for substring match and brevity (string length in chars).
+...
+```
+
+![Example AIConfig Eval output table](image-4.png)
+
+# Usage Guide - Custom Eval
+
+
+## Quick start:
+
+Please see [this notebook](https://github.com/lastmile-ai/aiconfig/blob/d79cf7dadf934b0ce09bc671cfde37aaecf05c1e/python/src/aiconfig/eval/custom_eval/examples/travel/travel_eval.ipynb#L4) for an example of running evaluation on an existing AIConfig. Make sure the AIConfig library is installed and then run the notebook cells in order.
```

Comment:
Make sure the AIConfig library is installed (`pip3 install -r PATH?TO/requirements.txt`)

Reply:
I think it would be good to separate out the bottom part to keep the README simpler.

Ex: "The notebook should be sufficient to teach you how things work, but if you're curious you can read <CustomEvalDetails.md> for more detailed understanding on Custom Eval guide."